src/libraries/Microsoft.Extensions.Options.DataAnnotations/src/DataAnnotationValidateOptions.cs

```diff
@@ -35,27 +35,29 @@ public DataAnnotationValidateOptions(string name)
         public ValidateOptionsResult Validate(string name, TOptions options)
         {
             // Null name is used to configure all named options.
-            if (Name == null || name == Name)
+            if (Name != null && !string.Equals(name, Name))
             {
-                var validationResults = new List<ValidationResult>();
-                if (Validator.TryValidateObject(options,
-                    new ValidationContext(options, serviceProvider: null, items: null),
-                    validationResults,
-                    validateAllProperties: true))
-                {
-                    return ValidateOptionsResult.Success;
-                }
+                // Ignored if not validating this instance.
+                return ValidateOptionsResult.Skip;
+            }
 
-                var errors = new List<string>();
-                foreach (ValidationResult r in validationResults)
-                {
-                    errors.Add($"DataAnnotation validation failed for members: '{string.Join(",", r.MemberNames)}' with the error: '{r.ErrorMessage}'.");
-                }
-                return ValidateOptionsResult.Fail(errors);
+            var validationResults = new List<ValidationResult>();
+            if (Validator.TryValidateObject(options,
+                new ValidationContext(options, serviceProvider: null, items: null),
+                validationResults,
+                validateAllProperties: true))
+            {
+                return ValidateOptionsResult.Success;
+            }
+
+            string typeName = options.GetType().Name;
+            var errors = new List<string>();
+            foreach (var result in validationResults)
+            {
+                errors.Add($"DataAnnotation validation failed for {typeName} members: '{string.Join(",", result.MemberNames)}' with the error: '{result.ErrorMessage}'.");
```

Comment:
What do you think about putting single quotes around the `'{typeName}'`? I think it is a bit easier to read that way.

Reply:
Makes sense to me. Especially if someone needs to act on the string value. Good indicators. Updated applied.